### DISCUSSION TITLE
How to reuse one function and called only 1 time with the same API but with different Query params?

### DISCUSSION THREAD

**Discussion by baiyyee** [fastapi/fastapi#3191] [Questions] [answered]
I have a function to get report details which called **get_report()**, and extra two functions both based on **get_report**() result to do something data analysis, they called **count_module()** and **count_priority()**, I really want to called get_report() 1 time for time saved, how to do it with FastAPI?

When send requests like:
.../showreport?report_type=module
.../showreport?report_type=priority

For Below code, looks it's called 2 times with function **get_report**, what I expect is called 1 time, because the two urls share the same **get_report** results, it's no need to called many times.

```python
from fastapi import FastAPI, Depends

app = FastAPI()

def get_report():
    fake_data = {}
    return fake_data

def count_module(datas):
    pass

def count_priority(datas):
    pass

@app.get("/showreport")
def show_report(report_type: str, datas: dict = Depends(get_report)):
    if report_type == "module":
        return count_module(datas=datas)

    elif report_type == "priority":
        return count_priority(datas=datas)
```

**Comment by Mause** [comment, marked as answer]
Caching of dependency results between requests isn't something that fastapi does, but you can do it yourself in various ways, such as using functools.lru_cache or the third party cachetools library (https://pypi.org/project/cachetools)

**Reply by baiyyee** [reply]
Got it, thanks.
